backend/lucene/src/main/java/org/hibernate/search/backend/lucene/search/impl/LuceneScopedIndexFieldComponent.java

```diff
@@ -8,19 +8,32 @@
 
 public class LuceneScopedIndexFieldComponent<T> {
 
-	private final T component;
-	private final LuceneCompatibilityChecker converterChecker;
-
-	public LuceneScopedIndexFieldComponent(T component, LuceneCompatibilityChecker converterChecker) {
-		this.component = component;
-		this.converterChecker = converterChecker;
-	}
+	private T component;
+	private LuceneCompatibilityChecker converterCompatibilityChecker = new LuceneSucceedingCompatibilityChecker();
+	private LuceneCompatibilityChecker analyzerCompatibilityChecker = new LuceneSucceedingCompatibilityChecker();
 
 	public T getComponent() {
 		return component;
 	}
 
+	public void setComponent(T component) {
```

Comment:
The idea would be the one to use the instance of `LuceneScopedIndexFieldComponent' as a collector, instead of using local variables.
Good idea to make the setters package visible only.